cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc

```diff
@@ -216,6 +216,11 @@ void OptimizationProblem::Solve(const std::vector<Constraint>& constraints,
     for (auto node_it = node_data_.begin(); node_it != node_data_.end();) {
       const int trajectory_id = node_it->id.trajectory_id;
       const auto trajectory_end = node_data_.EndOfTrajectory(trajectory_id);
+      if (frozen_trajectories.count(trajectory_id) != 0) {
+        // We skip frozen trajectories.
+        node_it = trajectory_end;
+        continue;
```

Comment:
Why not 'break'?

Reply:
We only want to skip this trajectory because it is frozen, but the next trajectory might need poses tied together by IMU again. In pure localization this is currently skipping trajectory 0 containing the map, continuing with trajectory 1 which is not frozen.